syft/frameworks/torch/tensors/interpreters/large_precision.py

```diff
@@ -0,0 +1,198 @@
+    large values by packing them in smaller ones.
+    Typically a user will require to enlarge a float number by fixing its precision
+        tensor.fix_prec()
+    These smaller values are of type `internal_type`. The large value is defined by `precision_fractional`
+
+    Check the tests to see how to play with the different parameters.
+    """
+
+    def __init__(
+        self,
+        owner=None,
+        id=None,
+        tags=None,
+        description=None,
+        base: int = 10,
+        precision_fractional=0,
+        internal_type=torch.int32,
+        verbose=False,
+    ):
+        """Initializes a LargePrecisionTensor.
+
+        Args:
+            owner (BaseWorker): An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id (str or int): An optional string or integer id of the LargePrecisionTensor.
+            tags (list): list of tags for searching.
+            description (str): a description of this tensor.
+            base (int): The base that will be used to to calculate the precision.
+            precision_fractional (int): The precision required by the caller.
+            internal_type (dtype): The large tensor will be stored using tensor of this type.
+        """
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.base = base
+        self.internal_type = internal_type
+        self.precision_fractional = precision_fractional
+        self.verbose = verbose
+
+    def _create_internal_tensor(self):
+        """Decompose a tensor into an array of numbers that represent such tensor with the required precision"""
+        # TODO refs_ok might not be needed as we are passing now torch.tensors here
+        # refs_ok is necessary to enable iterations of reference types.
+        # These big numbers are stored as objects in np
+        result = []
+        for x in np.nditer(self.child, flags=["refs_ok"]):
+            n = int(x.item() * self.base ** self.precision_fractional)
+            if self.verbose:
+                print(f"\nAdding number {n} for item {x.item()}\n")
+            result.append(self._split_number(n, internal_precision[self.internal_type]))
+        new_shape = self.child.shape + (len(max(result, key=len)),)
+        result = np.array(result).reshape(new_shape)
+        return torch.tensor(result, dtype=self.internal_type)
+
+    def get_class_attributes(self):
+        """
+        Specify all the attributes need to build a wrapper correctly when returning a response.
+        """
+        return {
+            "internal_type": self.internal_type,
+            "precision_fractional": self.precision_fractional,
+        }
+
+    @overloaded.method
+    def add(self, self_, other):
+        if self_.shape[-1] != other.shape[-1]:
+            return self._add_different_dims(other, self_)
+        else:
+            return self_ + other
```

Comment:
What happens when overflow occurs in one chunk of the data?

Reply:
We're dead here :) I was expecting to address the overflow when implementing the multiplication but it's true this can already be a problem with the addition.

In fact, this is probably the *main issue* with this representation of large numbers.

1) One possible solution could be to halve the storage precision. This way there will always be enough room to avoid an overflow.
This approach, however, will force to recheck the result in case the new number needs to be split again as soon a we concatenate more additions or multiplications.

2) Another option is to deal with these numbers the same way that a processor deals with numbers greater than the word they work with. This approach is slower but we know it works and we know there exists an implementation.

3) The third option would be to work directly with NumPy objects when operating. This would require splitting and restoring back the tensor that represents the large number with every operation. Probably this option is the most reasonable in terms of coding effort.

What do you think?

Reply:
Agreed on the 3rd, we shouldn't resent to use numpy when appropriate.
How does this PR relates with #2257 ? Is it just a fork of this one?

Reply:
#2257 was for showing how it could be with np. I will cancel it and keep working on this PR.

Reply:
Ok thanks for the update!